docs/source/developer/extension_migration.rst

```diff
@@ -0,0 +1,98 @@
+.. _extension_migration:
+
+JupyterLab 1.x to 2.x Extension Migration Guide
+------------------------------------------------
+
+This is a migration guide for updating extensions that support JupyterLab 1.x
+to work in JupyterLab 2.x. We will look at two examples of extensions that
+cover most of the APIs that extension authors might be using:
+
+- ``@jupyterlab/debugger`` migration pull request:
+  https://github.com/jupyterlab/debugger/pull/337/files
+
+- ``@jupyterlab/shortcutui`` migration pull request:
+  https://github.com/jupyterlab/jupyterlab-shortcutui/pull/53/files
+
+Upgrading library versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``@phosphor/*`` libraries that JupyterLab 1.x uses have been renamed to
+``@lumino/*``. Updating your ``package.json`` is straightforward. The easiest
+way to do this is to look in the
+`JupyterLab core packages code base <https://github.com/jupyterlab/jupyterlab/tree/master/packages>`__
+and to simply adopt the versions of the relevant libraries that are used
+there.
+
+.. figure:: extension_migration_dependencies_debugger.png
+   :align: center
+   :class: jp-screenshot
+   :alt: Updating the debugger extension's libraries in package.json
+
+   Updating the debugger extension's libraries in ``package.json``
+
+.. figure:: extension_migration_dependencies_shortcuts.png
+   :align: center
+   :class: jp-screenshot
+   :alt: Updating the shortcuts UI extension's libraries in package.json
+
+   Updating the shortcuts UI extension's libraries in ``package.json``
+
+In both cases, note that we are using the ``beta`` version of many libraries.
+This is to test the extensions against the JupyterLab 2.0 beta release before
+the final version is published. At the time you are upgrading your extension,
+the final versions of these packages may be published.
```

Comment:
Should there be an explicit sentence that would say something like: "You can directly bump to `^2.0.0`" ?

Reply:
Is that true? Does that work today?

Reply:
I meant if someone looks at the migration guide after the final versions of the packages have been published, they can use `^2.0.0` while the screenshots will be showing the `-beta` suffix.